src/app/frontend/common/components/roleref/component.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, Input} from '@angular/core';
+import {RoleRef} from '@api/backendapi';
+
+@Component({
+  selector: 'kd-role-ref-meta',
+  templateUrl: './template.html',
+})
+export class RoleRefComponent {
+  @Input() initialized = false;
+
+  private roleRef_: RoleRef;
```

Comment:
I don't think we need get/set here. It can simply be `@Input() roleRef: RoleRef = {} as RoleRef;` and then in `ngOnInit` method you can do something like `this.roleRef = this.roleRef === undefined ? {} as RoleRef : this.roleRef;`.

Reply:
okay, I will modifiy it later.